### ISSUE TITLE
Package request: Millennium

### ISSUE THREAD

**Issue by Krutonium** [NixOS/nixpkgs#325142]
**Project description**

An application that injects into your Steam Client, allowing things like themes and adding your own software to the overlay.

**Metadata**

* homepage URL: https://github.com/SteamClientHomebrew/Millennium
* source URL: https://github.com/SteamClientHomebrew/Millennium
* license: MIT
* platforms: Windows, Linux, Darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Krutonium** [comment]
Bumping because I still want this.

**Comment by keenanweaver** [comment]
Looks like some work has been done: https://github.com/kaeeraa/millennium-nix

**Comment by SmArtKar** [comment]
> Looks like some work has been done: [kaeeraa/millennium-nix](https://github.com/kaeeraa/millennium-nix?rgh-link-date=2025-02-12T14%3A53%3A53Z)

Non-functional and archived, looking into getting it to work currently.

**Comment by Sk7Str1p3** [comment]
@SmArtKar @keenanweaver I got some progress in my millennium fork, check it out

**Comment by LazyStability** [comment]
Is this still being worked on? Need any help?

**Comment by Sk7Str1p3** [comment]
Kinda abandoned packaging due to other projects but now I can continue. I will create new fork (or rather clean repository) (because I experienced some bugs then building on master )and if you want to help please open PR

**Comment by Superredstone** [comment]
are there any updates?

**Comment by Sk7Str1p3** [comment]
There's an issue with project structure I think? I'm not that experienced to package tool working on JavaScript, Python and C++. I haven't any experience with such mix, there was many problems. Sorry. 


I will try to do it one more time tho, in a few days

**Comment by Krutonium** [comment]
Regardless of how any of this turns out, @Sk7Str1p3 Thank you for taking the time to try :)

**Comment by Sk7Str1p3** [comment]
Checkout [tracking issue](https://github.com/SteamClientHomebrew/Millennium/issues/361) and my [fork](https://github.com/Sk7Str1p3/Millennium-nix), some progress has made so far, kind of stuck with CURL_OPENSSL error and patches, so please help if you want

**Comment by Sk7Str1p3** [comment]
Packaging is currently impossible because `pkgsStatic` are building with `musl` only, and millennium should be used only with glibc

**Comment by Sk7Str1p3** [comment]
Done. I will open PR in few hours. In future, I will provide NixOS/home-manager modules and plugins/themes packages so you can configure in declarative way

**Comment by Krutonium** [comment]
Can you tag this issue in the PR?
